### DISCUSSION TITLE
Typography: give lists and blockquotes `display: flow-root`?

### DISCUSSION THREAD

**Discussion by MichaelAllenWarner** [tailwindlabs/tailwindcss#5641] [Ideas]
If TailwindCSS Typography's `ul`, `ol`, or `blockquote` ends up against a left-floated element, then there's need for a "clearfix" (see sample images below). Giving those elements `display: flow-root` solves the problem. I wonder if there would be any downside to including this out of the box.

The trouble with these "clearfix" situations is that you wouldn't typically notice them in development—the problem only reveals itself once someone puts a list or blockquote unfortunately close to a left-floated item. So a "preemptive" fix is nice to have.

Of course, there are no _core_ Typography elements that get floated, but it's extremely common to float pieces in WYSIWYG contexts (especially images). If there are no detrimental side effects, this might be worth considering.

![Screen Shot 2021-09-29 at 10 47 28 AM](https://user-images.githubusercontent.com/44578578/135293228-045e9086-5dc9-48a5-bfa0-4f454d12d839.png)
![Screen Shot 2021-09-29 at 10 48 28 AM](https://user-images.githubusercontent.com/44578578/135293230-aa6573b6-0613-4d9e-8909-f7cebf1fb750.png)
![Screen Shot 2021-09-29 at 10 48 58 AM](https://user-images.githubusercontent.com/44578578/135293232-8c9635af-8390-438f-82af-05b276c59899.png)

**Comment by MichaelAllenWarner** [comment]
Well, `display: flow-root` on those elements does affect height and vertical-spacing, so this would _not_ work without further modification.

**Reply by MichaelAllenWarner** [reply]
I think the problem is that `display: flow-root` prevents margin-collapse.

**Comment by MichaelAllenWarner** [comment]
Something like this might work, "compensating" for the lack of margin-collapse:

```scss
ul:not(ul ul, ol ul),
ol:not(ul ol, ol ol),
blockquote {

  display: flow-root;

  > :first-child {
    margin-top: 0;
  }

  > :last-child {
    margin-bottom: 0;
  }
}
```

**Reply by MichaelAllenWarner** [reply]
Going a few layers "deeper" with the margin-zeroing (and doing away with the `:not()`s) seems to work pretty well:

```scss
ul,
ol,
blockquote {

  display: flow-root;

  > :first-child,
  > :first-child > :first-child,
  > :first-child > :first-child > :first-child {
    margin-top: 0;
  }

  > :last-child,
  > :last-child > :last-child,
  > :last-child > :last-child > :last-child {
    margin-bottom: 0;
  }
}
```

Not good core-Typography material, I'd say, but maybe someone will find this useful.

**Comment by Margret3** [comment]
The <blockquote> HTML element indicates that the enclosed text is an extended quotation. Usually, this is rendered visually by indentation (see Notes for how to change it). A URL for the source of the quotation may be given using the cite attribute, while a text representation of the source can be given using the <cite> element.

`blockquote {
    margin: 0;
}

blockquote p {
    padding: 15px;
    background: #eee;
    border-radius: 5px;
}

blockquote p::before {
    content: '\201C';
}

blockquote p::after {
    content: '\201D';
}
`

[skylightpaycard](https://www.skylightpaycard.us/)
